src/tool/hpcrun/write_data.c

```diff
@@ -175,7 +176,9 @@ lazy_open_data_file(core_profile_trace_data_t * cptd)
 
   int rank = hpcrun_get_rank();
 
-  compute_profile_name(rank, cptd);
+  if (cptd->id < 500){
```

Comment:
Why there's still this limitation?

Reply:
In this version threads with id < 500 are cpus, we can put a different thing in if clause like cptd->id_tuples == NULL

Reply:
Yeah, I would prefer the latter or an alternative check here. `500` CPU threads is quite limited.

Reply:
I have just added is_id_tuple_empty() check